pkg/operator/secretannotator/status/status.go

```diff
@@ -0,0 +1,87 @@
+package status
+
+import (
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorv1 "github.com/openshift/api/operator/v1"
+
+	"github.com/openshift/cloud-credential-operator/pkg/operator/constants"
+	"github.com/openshift/cloud-credential-operator/pkg/operator/utils"
+	"github.com/openshift/cloud-credential-operator/pkg/util/clusteroperator"
+)
+
+// SyncOperatorStatus computes the operator's current status and
+// creates or updates the ClusterOperator resource for the operator accordingly.
+func SyncOperatorStatus(kubeClient client.Client) error {
+	logger := log.WithField("controller", "secretannotator_status")
+	return clusteroperator.SyncStatus(kubeClient, logger)
+}
+
+func GetConditions(kubeClient client.Client, logger log.FieldLogger) ([]configv1.ClusterOperatorStatusCondition, error) {
+	conditions := []configv1.ClusterOperatorStatusCondition{}
+
+	mode, _, conflict, err := utils.GetOperatorConfiguration(kubeClient, logger)
+	if err != nil {
+		return conditions, err
+	}
+
+	// shouldn't happen with the server-side enforcement of the CRDs enum specification
+	if valid := utils.IsValidMode(mode); !valid {
+		conditions = append(conditions, configv1.ClusterOperatorStatusCondition{
+			Type:    configv1.OperatorDegraded,
+			Status:  configv1.ConditionTrue,
+			Reason:  constants.StatusModeInvalid,
+			Message: fmt.Sprintf("operator mode of %s is invalid", mode),
+		})
+	}
+
+	if conflict {
+		conditions = append(conditions, configv1.ClusterOperatorStatusCondition{
```

Comment:
If there is an invalid mode **and** a conflict, then the Degraded condition will be added twice.